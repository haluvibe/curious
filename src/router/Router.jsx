import React from "react";
import { Route, Switch, BrowserRouter } from "react-router-dom";
import PrivateRoute from "./PrivateRoute";
import App from "../components/App/App";
import FormPage from "../pages/Form";
import LoginPage from "../pages/Login";
import DefaultLayout from "../layouts/DefaultLayout";

export default () => 
  <BrowserRouter>
    <DefaultLayout>
      <Switch>
        <Route path="/" exact component={App} />
        <Route path="/login" exact component={LoginPage} />
        <PrivateRoute path="/form" exact component={FormPage} />
      </Switch>
    </DefaultLayout>
  </BrowserRouter>
;
