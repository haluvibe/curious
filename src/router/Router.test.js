import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

import Router from "./Router";

test("Router component should render as expected", () => {
  const component = shallow(<Router />),
    tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
