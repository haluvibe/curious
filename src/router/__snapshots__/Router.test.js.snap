// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router component should render as expected 1`] = `
<BrowserRouter>
  <WithStyles(DefaultLayout)>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/login"
      />
      <Connect(PrivateRoute)
        component={[Function]}
        exact={true}
        path="/form"
      />
    </Switch>
  </WithStyles(DefaultLayout)>
</BrowserRouter>
`;
